src/helpers/FmspcTcbHelper.sol

```diff
@@ -184,15 +181,31 @@ contract FmspcTcbHelper {
         bool pceidFound;
         bool evaluationFound;
         bool idFound;
-        bool allFound;
+        bool tcbLevelsFound;
+        bool tdxModuleFound;
+        bool tdxModuleIdentitiesFound;
+    }
+
+    function parseTcbString(string calldata tcbInfoStr) external pure returns (
+        TcbInfoBasic memory tcbInfo,
+        string memory tcbLevelsString,
+        string memory tdxModuleString,
+        string memory tdxModuleIdentitiesString
+    ) {
+        JSONParserLib.Item memory root = JSONParserLib.parse(tcbInfoStr);
+        JSONParserLib.Item[] memory tcbInfoObj = root.children();
+
+        // move flags to memory to avoid stack too deep
+        ParseFlag memory f;
```

Comment:
f can just be a uint256.

you set the bits by doing a 

f = f | 2**(bit_pos) // value is constant.

the final check will be f == 2 ** 12 - 1 // again a constant value check